src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -24519,6 +25192,642 @@ static void M1()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class CClass<TClass> where TClass : class
+{
+    public TClass P2 { get; [DisallowNull] set; } = null!;
+    public TClass? P3 { get; [DisallowNull] set; } = null;
+}";
+            var comp = CreateNullableCompilation(new[] { DisallowNullAttributeDefinition, source });
+            comp.VerifyDiagnostics(
+                // (4,30): error CS0592: Attribute 'DisallowNull' is not valid on this declaration type. It is only valid on 'property, indexer, field, parameter' declarations.
+                //     public TClass P2 { get; [DisallowNull] set; } = null!;
+                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "DisallowNull").WithArguments("DisallowNull", "property, indexer, field, parameter").WithLocation(4, 30),
+                // (5,31): error CS0592: Attribute 'DisallowNull' is not valid on this declaration type. It is only valid on 'property, indexer, field, parameter' declarations.
+                //     public TClass? P3 { get; [DisallowNull] set; } = null;
+                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "DisallowNull").WithArguments("DisallowNull", "property, indexer, field, parameter").WithLocation(5, 31)
+                );
+        }
+
+        [Fact]
+        public void DisallowNull_Property_OnGetter()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class CClass<TClass> where TClass : class
+{
+    public TClass P2 { [DisallowNull] get; set; } = null!;
+    public TClass? P3 { [DisallowNull] get; set; } = null;
+}";
+            var comp = CreateNullableCompilation(new[] { DisallowNullAttributeDefinition, source });
+            comp.VerifyDiagnostics(
+                // (4,25): error CS0592: Attribute 'DisallowNull' is not valid on this declaration type. It is only valid on 'property, indexer, field, parameter' declarations.
+                //     public TClass P2 { [DisallowNull] get; set; } = null!;
+                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "DisallowNull").WithArguments("DisallowNull", "property, indexer, field, parameter").WithLocation(4, 25),
+                // (5,26): error CS0592: Attribute 'DisallowNull' is not valid on this declaration type. It is only valid on 'property, indexer, field, parameter' declarations.
+                //     public TClass? P3 { [DisallowNull] get; set; } = null;
+                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "DisallowNull").WithArguments("DisallowNull", "property, indexer, field, parameter").WithLocation(5, 26)
+                );
+        }
+
+        [Fact]
+        public void DisallowNull_Property_NoGetter()
+        {
+            // Warn on misused nullability attributes? https://github.com/dotnet/roslyn/issues/36073
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class COpen<TOpen>
+{
+    [DisallowNull]public TOpen P1 { set; }
```

Comment:
>set; [](start = 36, length = 4)

Consider adding implementation so there is no error, assuming the error is not what is being tested here.